a-short-introduction-to-kubernetes/index.md

```diff
@@ -0,0 +1,372 @@
+
+One of the reasons is then you create an object, like a Deployment, a random string is used for the pod it creates. So you to communicate with it, you need the metadata to identify it.
+
+Another, extremely useful metadata is annotations, but that is out of the scope for this post.
+
+## Containers
+
+Containers are the Docker images you deploy to your K8s cluster. There are mainly two kind of containers, initContainers and containers. initContainers runs before your containers start, like if you need to set permissions, or do some task like updating your Directus schema etc.
+
+## Environment variables
+
+Environment variables could be added as part of your object, like the public url for Directus.
+
+```yaml
+apiVersion: v1
+kind: Pod
+metadata:
+  name: directus-app
+  labels:
+    app: directus
+spec:
+  containers:
+  - name: directus
+    image: directus/directus:10.8.3
+    ports:
+    - containerPort: 80
+    env:
+      - name: PUBLIC_URL
+        value: https://directus.app
+```
+
+You could also use Secrets and use with for setting environment variables, that I will not cover here.
+
+## Pods
+
+A pod is the smallest deployable compute object in K8s, like the example above. A pod is what is running one or several docker containers, a pod could be Directus, MySQL or Redis. If your pod crashes, it's dead and doesn't restart - it's like when you start a docker container with `docker run directus` and suddenly something gets weird, or your resources run out. So you need something to handle pods with, and the logic they are started, with config and volumes, there is where Deployments and StatefulSets come in.
+
+## Deployments
+
+A Deployment manages a set of Pods to run an application workload, usually one that doesn't maintain state (coming that later). A Deployment is a way to describe the pod(s) you want to run, if you need to mount volumes (for storing data) or config (like the config file for your Directus Deployment), also you add logic for resources is going to use (RAM, and CPU). So, with a Deployment, you create pods. Each Deployment, creates normally one pod (which, like state above, could have many containers).
+
+A Deployment could "update" your pod - and what happens is that K8s takes down the existing pod, and replaces with a new one, like when updating your Directus instance, or adding new config or env variables.
+
+```yaml
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: directus-Deployment
+  labels:
+    app: directus
+spec:
+  selector:
+    matchLabels:
+      app: directus
+  template:
+    metadata:
+      labels:
+        app: directus
+    spec:
+      containers:
+      - name: directus
+        image: directus:1.32.3
+        ports:
+          - containerPort: 80
+        env:
+          - name: PUBLIC_URL
+            value: https://directus.app
+```
+
+## StatefulSets
```

Comment:
```suggestion
## `StatefulSets`
```

Can you wrap all thesein backticks? Otherwise the spellchecker is gonna be upset.